docs/functions.md

```diff
@@ -0,0 +1,209 @@
+
+    It is intended to be used for brevity, increased productivity and the reduction of finger strain due to repetitive typing.
+
+**Python API**
+```python
+class GearsBuilder(reader='KeysReader', defaultArg='*', desc=None)
+```
+_Arguments_
+
+* _reader_: the function's [reader](readers.md)
+* _defaultArg_: An optional argument that the reader may need. These are usually a key's name, prefix, glob-like or a regular expression. Its use depends on the function's reader type and action.
+* _desc_: an optional description
+
+**Examples**
+```python
+# Here's the default context builder being run
+GearsBuilder().run()
+
+# You can also do this
+gb = GB()
+gb.register()
+```
+
+## Actions
+An action is a special type of operation. It is always the function's final step.
+
+### Run
+The **Run** action runs a function as a batch. The function is executed once and exits once the data is exhausted by its reader.
+
+Trying to run more than one function in the same execution will fail with an error.
+
+!!! example "Multiple executions error"
+    ```
+    127.0.0.1:30001> RG.PYEXECUTE "GB().run()\nGB().run()"
+    (error) [... 'spam.error: Can not run more then 1 executions in a single script']
+    ```
+
+!!! important "Execution is always asynchronous"
+    Batch functions are **always executed asynchronously** by the RedisGears engine. That means that they are run in a background thread rather than by the main process of the Redis server.
+
+**Python API**
+```python
+class GearsBuilder.run(arg=None, convertToStr=True, collect=True)
+```
+
+_Arguments_
+
+* _arg_: An optional argument that's passed to the reader as its _defaultArg_. It means the following:
+    * A glob-like pattern for the [KeysReader](readers.md#keysreader) and [KeysOnlyReader](readers.md#keysonlyreaders) readers
+    * A key name for the [StreamReader](readers.md#streamreader) reader
+* _convertToStr_: when `True` adds a [map](operations.md#map) operation to the flow's end that stringifies records
+* _collect_: when `True` adds a [collect](operations.md#collect) operation to flow's end
+
+**Examples**
+```python
+# Runs the function
+GB() \
+  .run()
+```
+
+### Register
+The **Register** action registers a function as an event handler. The function is executed each time an event arrives. Each time it is executed, the function operates on the event's data and once done is suspended until its future invocations by new events.
+
+**Python API**
+```python
+class GearsBuilder.register(regex='*', mode='async', batch=1, duration=0,
+  eventTypes=None, keyTypes=None, onRegistered=None, onFailedPolicy="continue",
+  onFailedRetryInterval=1)
+```
+
+_Arguments_
+
+* _regex_: An optional argument that's passed to the reader as its _defaultArg_. It means the following:
+    * A key prefix for the [KeysReader](readers.md#keysreader) and [KeysOnlyReader](readers.md#keysonlyreaders) readers
+    * A key name for the [StreamReader](readers.md#streamreader) reader
```

Comment:
On stream reader here prefix is also supported ..

Reply:
Fixed via 079f280